subgraphs/uniswap-forks/protocols/sushiswap/src/common/handlers/handleReward.ts

```diff
@@ -62,26 +66,40 @@ export function handleReward(
   }
 
   // Get necessary values from the master chef contract to calculate rewards
-  let poolInfo = poolContract.poolInfo(pid);
-  masterChefPool.poolAllocPoint = poolInfo.value1;
-
-  // Mutliplier including block mulitplier
-  let fullMultiplier = poolContract
-    .getMultiplier(masterChefPool.lastRewardBlock, event.block.number)
-    .div(masterChefPool.lastRewardBlock.minus(event.block.number));
+  let getPoolInfo = poolContract.try_poolInfo(pid);
+  if (!getPoolInfo.reverted) {
+    let poolInfo = getPoolInfo.value;
+    masterChefPool.poolAllocPoint = poolInfo.value1;
+  }
 
-  // Divide out the block multiplier so only the bonus multiplier is left
-  masterChefPool.multiplier = fullMultiplier.div(
-    masterChefPool.lastRewardBlock.minus(event.block.number)
+  // Get the bonus multiplier if it is applicable
+  let getMuliplier = poolContract.try_getMultiplier(
+    event.block.number.minus(BIGINT_ONE),
+    event.block.number
   );
-  masterChef.totalAllocPoint = poolContract.totalAllocPoint();
+  if (!getMuliplier.reverted) {
+    masterChefPool.multiplier = getMuliplier.value;
+  }
+
+  // Get the total allocation for all pools
+  let getTotalAlloc = poolContract.try_totalAllocPoint();
+  if (!getTotalAlloc.reverted) {
+    masterChef.totalAllocPoint = getTotalAlloc.value;
+  }
 
   // Address where allocation is moved to over time to reduce inflation
-  let masterPoolAllocPID45 = poolContract.poolInfo(BigInt.fromI32(45)).value1;
+  let getPoolInfo45 = poolContract.try_poolInfo(BigInt.fromI32(45));
```

Comment:
is this a burner address? might worth adding a bit more comments to explain

Reply:
Yeah - this is a burner address. I'll put more comments in.